src/InteractiveWindow/EditorTest/InteractiveWindowTests.cs

```diff
@@ -629,12 +629,15 @@ public void CheckHistoryPrevious()
             Assert.Equal(V, GetTextFromCurrentLanguageBuffer);
         }
 
-        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/4121")]
-        public void CheckHistoryAfterResetButtonClick()
+        [Fact]
+        public void CheckHistoryPreviousAfterReset()
         {
-            Task.Run(() => Window.Operations.ResetAsync(initialize: true)).PumpingWait();
-            Task.Run(() => Window.Operations.HistoryPrevious()).PumpingWait();
-            Assert.Equal("#reset", GetTextFromCurrentLanguageBuffer);
+            const string V = "#reset";
```

Comment:
Changed the name. Sorry I was just lazy and it was late :-)
Yes it would be worthwhile to test that history only has a single entry. I have a issue (4307) assigned to me to write more tests for these bugs. I will include this as well.